### DISCUSSION TITLE
how to read file content(has template literal)  into  template literal ?

### DISCUSSION THREAD

**Discussion by dili-world** [denoland/deno#10711] [Q&A]
### code 
hello.html
```html
hello ${username}
```

main.ts
```js
let username = "john";
const data = await Deno.readFile("hello.html");
let bodyContent = decoder.decode(data);
bodyContent = `${bodyContent}`;// i want to let the bodyContent put into template literal
console.log(bodyContent); //output is "hello ${username}",  how to output "hello john" use template literal?

```

or 

```
let username = "john";
let fileContent = "hello ${username}";
let template= `${fileContent}`; 
console.log(template);// i mean the output is "hello john" not "hello ${username}"
```

**Comment by wperron** [comment]
I'm not exactly sure what you're trying to do here? `decoder.decode` already converts `data` from a `Uint8Array` to `string`, putting it in a string template literal like doesn't do anything in that context. Is there more info you could give?

**Reply by dili-world** [reply]
decoder.decode's Uint8Array is the html file's content "hello ${username}",  i want a way to get "hello john"

**Reply by dili-world** [reply]
use template literal

**Reply by dili-world** [reply]
let content = "hello ${username}"; // this is the file content put in , then i want to use template literal , content = `${content}`;  i want it output "hello john"

**Reply by dili-world** [reply]
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

**Reply by wperron** [reply]
afaik you can't read a template literal from a file; template literals are a form of functions. You might want to look into something like https://github.com/handlebars-lang/handlebars.js instead

**Reply by kitsonk** [reply]
Or you would need to eval it in a particular context. Ultimately it is pretty unsafe way of doing things and using something like an existing templating language is much safer.

Ultimately this isn't Deno specific and this StackOverflow pretty much answers the question more thoroughly: https://stackoverflow.com/questions/29182244/convert-a-string-to-a-template-string
